File: Backend/LinkCare/src/main/java/com/ssafy/linkcare/character/dto/CharacterStatusDto.java
package com.ssafy.linkcare.character.dto;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class CharacterStatusDto {
    private Long characterId;
    private String name;
    private String description;
    private String baseImageUrl;
    private String animatedImageUrl;
    private boolean isUnlocked;
    private boolean isMain;
}

File: Frontend/app/src/main/java/com/a307/linkcare/feature/character/data/model/dto/CharacterStatusDto.kt
package com.a307.linkcare.feature.character.data.model.dto

import com.a307.linkcare.feature.character.domain.model.CharacterManifest
import kotlinx.serialization.Serializable

@Serializable
data class CharacterStatusDto(
    val characterId: Long,
    val name: String,
    val description: String? = null,
    val imageUrls: List<String> = emptyList(),
    val isUnlocked: Boolean,
    val isMain: Boolean,
    val manifest: CharacterManifest? = null
)